README.md

```diff
@@ -99,15 +100,132 @@ use, and what the relevant image actually builds.
+mindful of that possibility: 
+
+```
+λ> let plus1 :: Int ⊸ Int; plus1 x = x + 1
+
+<interactive>:12:31: error:
+    • Couldn't match expected weight ‘1’ of variable ‘x’ with actual weight ‘ω’
+    • In an equation for ‘plus1’: plus1 x = x + 1
+```
+
+This unfortunately applies to _all of prelude_ (`+` is the offender
+here), which means you might see this a few times.
+
+
+Currently the type checker does not check linearity when not asked to do so,
+hence it is a requirement to explicitly declare the types of linear functions.
+Otherwise, they are assumed to be unrestricted which can give you errors from
+using functions that are _implemented_ linearly, but not declared as such.
+In particular, functions in `where`-clauses are easy to forget to annotate
+(`-XPartialTypeSignatures` or `-XScopedTypeVariables` are useful for this),
+but lambdas are inferred properly.
+
+
+## Calling them
+By design, there is no obligations put on the caller of a linear function. This
+means any linear function can take unrestricted arguments, which is great
+since it makes all functions "linearly polymorphic". However, passing a
+_linear_ variable as an _unrestricted_ argument is illegal and will give you
+an error, as shown above. The key takeaway here is that any linear function
+can also act as an unrestricted one; just pass it unrestricted arguments.
+
+
+## Unimplemented features & known limitations
+There are some parts of Haskell that are not yet implemented linearly, meaning
+you cannot use them with linear variables, most
+notably `case` and `let`:
+
+```
+λ> let f :: a ⊸ a; f x = case x of x -> x
+
+<interactive>:27:19: error:
+    • Couldn't match expected weight ‘1’ of variable ‘x’ with actual weight ‘ω’
+    • In an equation for ‘f’: f x = case x of { x -> x }
+
+
+λ> let f :: a ⊸ a; f x = let y = x in y
+
+<interactive>:28:19: error:
+    • Couldn't match expected weight ‘1’ of variable ‘x’ with actual weight ‘ω’
+    • In an equation for ‘f’: f x = let y = x in y
+```
+
+`@`-patterns also do not work as expected:
+
+```
+λ> let f :: a ⊸ a; f y@x = y
+
+<interactive>:32:21: error:
+    • Couldn't match expected weight ‘1’ of variable ‘x’ with actual weight ‘0’
+    • In an equation for ‘f’: f y@x = y
+```
+
+Other than that, pattern synonyms are also not compatible with linear types
+(and probably incorrect). These are the current _known_ limitations; there may
+be more but it has not been thoroughly tested yet.
+
+
+## Bugs
+There is currently one particular bug causing headaches, where linear and
+non-linear arrows get equalised under certain conditions. It happens when you
+pass a non-linear function where a linear function is expected, and the
+linearity checker happily accepts this, making stuff like this look fine while
+it really is not:
+
+```
+λ> skip :: a ⊸ (); skip = const ()
+```
+
+There are probably more bugs; if you find something that definitely looks off,
+we would appreciate a well-documented bug report to this repo.
```

Comment:
Add link to the issue page: https://github.com/tweag/ghc/issues .

Use the word "repository" rather than "repo.